option.go

```diff
@@ -119,6 +119,18 @@ func EqualFuncs(b bool) Option {
 	}}
 }
 
+// ZeroFields copies values and sets the specified fields to their zero value.
+func ZeroFields[T any](fields ...string) Option {
+	return Transform(func(v T) any {
+		rv := reflect.ValueOf(&v)
+		for _, name := range fields {
+			fv := rv.Elem().FieldByName(name)
+			fv.Set(reflect.Zero(fv.Type()))
```

Comment:
It would be good to check that the fields exist before ZeroFields returns. That way we are guaranteed to catch the error. (If there are no T values being compared, we would never run the transform, so this wouldn't have a chance to panic.) We would also perhaps find the error earlier (e.g. if this option is being defined in a global during package init).

Something along these lines:

```go
func ZeroFields[T any](fields ...string) Option {
	t := reflect.TypeOf((*T)(nil).Elem()
	for _, name := range fields {
		if _, ok := t.FieldByName(name); !ok {
			panic("diff: field not found: " + name)
		}
	}
	return Transform(func(v T) any {
		...
	})
}
```